rapidy/_annotation_container.py

```diff
@@ -1,23 +1,76 @@
 import inspect
 from abc import ABC, abstractmethod
+from enum import Enum
 from types import FunctionType
-from typing import Any, Dict, Iterator, Set, Type, Union
+from typing import Any, Dict, Iterator, Optional, Set, Type, Union
 
-from aiohttp.typedefs import Middleware
 from aiohttp.web_request import Request
+from typing_extensions import get_args
 
-from rapidy._annotation_extractor import extract_handler_attr_annotations, UnknownParameterError
-from rapidy._client_errors import _create_handler_info_msg, ExtractError
+from rapidy._annotation_extractor import extract_handler_attr_annotations, NotParameterError
+from rapidy._client_errors import _create_handler_attr_info_msg, _create_handler_info_msg, ExtractError
 from rapidy._validators import validate_request_param_data
 from rapidy.fields import ModelField
 from rapidy.request_params import create_param_model_field_by_request_param, ParamFieldInfo, ParamType, ValidateType
-from rapidy.typedefs import ValidateReturn
+from rapidy.typedefs import Handler, MethodHandler, Middleware, NoArgAnyCallable, ValidateReturn
+
+
+class HandlerEnumType(str, Enum):
+    func = 'func'
+    method = 'method'
+    middleware = 'middleware'
```

Comment:
```suggestion
    func = enum.auto()
    method = enum.auto()
    middleware = enum.auto()
```

Reply:
fix

Reply:
What do you mean? It's just generic method to generate a value if you change a key

Reply:
<img width="601" alt="image" src="https://github.com/daniil-grois/rAPIdy/assets/82909417/36cdaa70-b569-4dcf-9df5-5f85079d6706">